Moviebuff/Views/MovieListView.swift

```diff
@@ -0,0 +1,75 @@
+
+import SwiftUI
+
+struct MovieListView: View {
+    @ObservedObject private var nowPlayingState = MovieViewModel()
+    @ObservedObject private var bestDramasState = MovieViewModel()
+    @ObservedObject private var topRatedState = MovieViewModel()
+    @ObservedObject private var popularState = MovieViewModel()
+    
+    var body: some View {
+        NavigationView {
+            List {
+                Group{
+                    if nowPlayingState.movies != nil {
+                        PosterCarouselView(title: "Now Playing", movies: nowPlayingState.movies!, viewModel: nowPlayingState)
+                    }else {
+                        LoadingView()
+                    }
+                }
+                .listRowInsets(EdgeInsets(top: 16, leading: 0, bottom: 16, trailing: 0))
+                Group {
```

Comment:
Another approach to all this repetition — especially as your app may become more dynamic — would be to use an array of objects that represent the rows that you want, and then iterate through them. You could even include a `title` string in your `MovieViewModel`, and store an array of models that you want to be displayed. You could also have that model decide whether it wants the Poster or the Backdrop view type.

Lots of ways to do this — I'm just trying to help you think of ways to be more dynamic, which is an approach that I personally like using a lot.